src/proxy_handler.cpp

```diff
@@ -0,0 +1,206 @@
+                                                         Response* response, int count) {
+  if (count >= MAX_REDIRECTS) {
+    return RequestHandler::INVALID_CONFIG;
+  }
+
+  try {
+    std::cout << "Host: " << proxy_host_ << std::endl;
+    std::cout << "Port: " << proxy_port_ << std::endl;
+    std::string body_;
+
+    boost::asio::io_service io_service;
+    
+    // Get a list of endpoints corresponding to the server name.
+    boost::asio::ip::tcp::resolver resolver(io_service);
+    boost::asio::ip::tcp::resolver::query query(proxy_host_, proxy_port_, boost::asio::ip::resolver_query_base::numeric_service);
+    boost::asio::ip::tcp::resolver::iterator endpoint_iterator = resolver.resolve(query);
+    boost::asio::ip::tcp::resolver::iterator end;
+    boost::system::error_code conn_error = boost::asio::error::host_not_found;
+    boost::asio::ip::tcp::socket socket (io_service);
+
+    while (conn_error && endpoint_iterator != end) {
+      socket.close();
+      socket.connect(*endpoint_iterator++, conn_error);
+    }
+
+    // Try each endpoint until we successfully establish a connection.
+
+    // Form the rrequest. We specify the "Connection: close" header so that the
+    // server will close the socket after transmitting the response. This will
+    // allow us to treat all data up until the EOF as the content.
+    boost::asio::streambuf request_;
+    std::ostream request_stream(&request_);
+    std::string redirect_uri;
+    std::string orig_uri = request.uri();
+    if (uri_prefix_.size() == 1) {
+      redirect_uri = orig_uri;
+    }
+    else {
+      if (uri_prefix_.size() + 1 >= orig_uri.size())
+        redirect_uri = "/";
+      else 
+        redirect_uri = orig_uri.substr(uri_prefix_.size());
+    }
+
+    if (!redirect_uri_.empty()) {
+      redirect_uri = redirect_uri_;
+    }
+    //TODO: Change to "/"
+    request_stream << "GET " + redirect_uri + " HTTP/1.0\r\n";
+    request_stream << "Host: " << proxy_host_ << "\r\n";
+    request_stream << "Accept: */*\r\n";
+    request_stream << "Connection: close\r\n\r\n";
+
+    // Send the request.
+    boost::asio::write(socket, request_);
+
+    // Read the response status line. The response streambuf will automatically
+    // grow to accommodate the entire line. The growth may be limited by passing
+    // a maximum size to the streambuf constructor.
+    boost::asio::streambuf response_;
+    boost::asio::read_until(socket, response_, "\r\n");
+
+    // Check that response is OK.
+    std::istream response_stream(&response_);
+    std::string http_version;
+    response_stream >> http_version;
+    unsigned int status_code;
+    response_stream >> status_code;
+    std::string status_message;
+    std::getline(response_stream, status_message);
+    if (!response_stream || http_version.substr(0, 5) != "HTTP/") {
+      std::cout << "Invalid response\n";
+      return RequestHandler::INVALID_CONFIG;
+    }
+    if (status_code != 200 && status_code != 302) {
```

Comment:
Sorry to be pedantic, but please maintain consistent if-else block structure. There must always be a new line after an ending }. else must be on the same line as the } belonging to the if-statement.